pkg/apis/hive/v1alpha1/validating-webhooks/selector_syncset_validating_admission_hook.go

```diff
@@ -161,6 +162,18 @@ func (a *SelectorSyncSetValidatingAdmissionHook) validateCreate(admissionSpec *a
 		}
 	}
 
+	if invalid, ok := checkSecretResources(newObject.Spec.SecretResources); !ok {
+		message := fmt.Sprintf("Failed validation: Invalid secret resource kind detected: %s. SecretResources may only contain references to kind: secret.", invalid)
+		contextLogger.Infof(message)
+		return &admissionv1beta1.AdmissionResponse{
+			Allowed: false,
+			Result: &metav1.Status{
+				Status: metav1.StatusFailure, Code: http.StatusBadRequest, Reason: metav1.StatusReasonBadRequest,
```

Comment:
I would expect this to be `metav1.StatusReasonInvalid` and for the details to indicate which field was invalid. See https://github.com/kubernetes/apimachinery/blob/c0632ccbde11a13044c0f41b7e3c588f16ecda8b/pkg/api/errors/errors.go#L222-L245.